test/extended/images/rails_ex.go

```diff
@@ -0,0 +1,83 @@
+package images
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"k8s.io/kubernetes/test/e2e"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("images: parallel: test ruby images with rails-ex repo", func() {
+	defer g.GinkgoRecover()
+	var (
+		repoName             = "rails"
+		templateURL          = "https://raw.githubusercontent.com/openshift/rails-ex/master/openshift/templates/rails-postgresql.json"
+		buildConfigName      = "rails-postgresql-example"
+		serviceName          = "rails-postgresql-example"
+		deploymentConfigName = "rails-postgresql-example"
+		expectedString       = "Listing articles"
+		appPath              = "/articles"
+		oc                   = exutil.NewCLI(repoName+"-repo-test", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("Waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeREST().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.Describe("Building "+repoName+" app from new-app", func() {
+		g.It(fmt.Sprintf("should build a "+repoName+" image and run it in a pod"), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			g.By(fmt.Sprintf("calling oc new-app with the " + repoName + " example template"))
+			err := oc.Run("new-app").Args("-f", templateURL).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("starting a build")
+			buildName, err := oc.Run("start-build").Args(buildConfigName).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("expecting the build is in the Complete phase")
+			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName, exutil.CheckBuildSuccessFunc, exutil.CheckBuildFailedFunc)
+			if err != nil {
+				logs, _ := oc.Run("build-logs").Args(buildName).Output()
+				e2e.Failf("build failed: %s", logs)
+			}
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("expecting the deployment to be complete")
+			err = exutil.WaitForADeployment(oc.KubeREST().ReplicationControllers(oc.Namespace()), deploymentConfigName, exutil.CheckDeploymentCompletedFunc, exutil.CheckDeploymentFailedFunc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("expecting the service is available")
+			serviceIP, err := oc.Run("get").Args("service", serviceName).Template("{{ .spec.clusterIP }}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(serviceIP).ShouldNot(o.Equal(""))
+
+			g.By("expecting an endpoint is available")
+			err = oc.KubeFramework().WaitForAnEndpoint(serviceName)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			resp, err := http.Get("http://" + serviceIP + ":8080/articles")
```

Comment:
sure (fyi this is generally being refactored in other ways,  so not worth much more reviewing right now)